test/rules/missing_whitespace_between_adjacent_strings.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N missing_whitespace_between_adjacent_strings`
+
+f(o) {
+  f('long line' // LINT
+    'is long');
+  f('long $f line' // LINT
+    'is long');
+  f('long line' ' is long'); // OK
+  f('long line ' 'is long'); // OK
+  f('long $f line ' 'is long'); // OK
+  f('longLineWithoutSpaceCouldBe' 'AnURL'); // OK
+
+  f(RegExp('(\n)+' '(\n)+' '(\n)+')); // OK
+  matches('(\n)+' '(\n)+' '(\n)+'); // OK
+}
```

Comment:
I think we should not produce a lint for code like `'your' '$string' 'here'` because we can't know that the value of `string` doesn't start or end with a space.

Reply:
`'your' '$string' 'here'` will not be linted because the lint is trigger only if the _current string contains a whitespace.

However a lint will be triggered for `'yo ur' '$string' 'here'` and that's fine IMHO. The user can change his code to `yo ur${string}here`.

In practice (on flutter codebase) there were only one place I add a similar case were the variable carried a trailing space. If we hadn't lint those kind of case we had missed around 10 real problems.

Reply:
That's reasonable. We can see whether it causes issues in other code bases and adjust it if necessary.